src/dataflow/src/sink/tail.rs

```diff
@@ -94,6 +94,12 @@ fn tail<G>(
 ) where
     G: Scope<Timestamp = Timestamp>,
 {
+    // `active_worker` indicates the index of the worker that receives all data.
+    // This should line up with the exchange just below, but we test in the implementation
+    // just to be certain.
+    let scope = sinked_collection.scope();
+    use differential_dataflow::Hashable;
+    let active_worker = (sink_id.hashed() as usize) % scope.peers() == scope.index();
```

Comment:
nit: I don't like to rely on the internals of exchange here, but I also don't see a better solution right now. And I agree that it's better to emit progress even before receiving any data!

Not a very useful comment, overall... 😅

Reply:
I agree. I could stabilizes timely's exchange pact as being a specific distribution over workers, e.g. that it sends 0 .. workers to the specified worker, and we could fix that up. But, .. it seems most important to get progress information out before data occurs. Another option is to have all workers participate, and then progress deduplication after the fact.

Reply:
I think it's good to merge as is, once ci agrees. 👍 

By the way, we also have this in at least one other place: https://github.com/MaterializeInc/materialize/blob/2f79b0237ec851bf389358b27a397e6420dd06e3/src/dataflow/src/sink/kafka.rs#L137. (I was also skeptical when it was added there.)